fe/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -0,0 +1,275 @@
+                // if column type is Date, format partition name as yyyyMMdd
+                // if column type is DateTime, format partition name as yyyyMMddHHssmm
+                // scheduler time should be record even no partition added
+                createOrUpdateRuntimeInfo(olapTable.getName(), LAST_SCHEDULER_TIME, TimeUtils.getCurrentFormatTime());
+                RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) olapTable.getPartitionInfo();
+                Column partitionColumn = rangePartitionInfo.getPartitionColumns().get(0);
+                String partitionFormat;
+                try {
+                    partitionFormat = DynamicPartitionUtil.getPartitionFormat(partitionColumn);
+                } catch (DdlException e) {
+                    recordFailedMsg(olapTable.getName(), e.getMessage());
+                    continue;
+                }
+
+                Calendar calendar = Calendar.getInstance();
+                TableProperty tableProperty = olapTable.getTableProperty();
+                DynamicPartitionProperty dynamicPartitionProperty = tableProperty.getDynamicPartitionProperty();
+
+                for (int i = 0; i <= dynamicPartitionProperty.getEnd(); i++) {
+                    String dynamicPartitionPrefix = dynamicPartitionProperty.getPrefix();
+                    String prevBorder = DynamicPartitionUtil.getPartitionRange(dynamicPartitionProperty.getTimeUnit(),
+                            i, (Calendar) calendar.clone(), partitionFormat);
+                    String partitionName = dynamicPartitionPrefix + DynamicPartitionUtil.getFormattedPartitionName(prevBorder);
+
+                    // continue if partition already exists
+                    String nextBorder = DynamicPartitionUtil.getPartitionRange(dynamicPartitionProperty.getTimeUnit(),
+                            i + 1, (Calendar) calendar.clone(), partitionFormat);
+                    PartitionValue lowerValue = new PartitionValue(prevBorder);
+                    PartitionValue upperValue = new PartitionValue(nextBorder);
+                    PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+                    RangePartitionInfo info = (RangePartitionInfo) (partitionInfo);
+                    boolean isPartitionExists = false;
+                    for (Range<PartitionKey> partitionKeyRange : info.getIdToRange().values()) {
+                        // only support single column partition now
+                        try {
+                            PartitionKey lowerBound = PartitionKey.createPartitionKey(Collections.singletonList(lowerValue), Collections.singletonList(partitionColumn));
+                            PartitionKey upperBound = PartitionKey.createPartitionKey(Collections.singletonList(upperValue), Collections.singletonList(partitionColumn));
+                            Range<PartitionKey> addPartitionKeyRange = Range.closedOpen(lowerBound, upperBound);
```

Comment:
you can create the `addPartitionKeyRange` outside the for loop, no need to create it every time